applications/OptimizationApplication/tests/test_optimization_utils.py

```diff
@@ -0,0 +1,61 @@
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+
+# Import KratosUnittest
+import KratosMultiphysics.KratosUnittest as kratos_unittest
+
+class TestOptimizationUtils(kratos_unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.model = Kratos.Model()
+        cls.model_part = cls.model.CreateModelPart("test")
+        cls.model_part.AddNodalSolutionStepVariable(Kratos.DENSITY)
+        cls.model_part.AddNodalSolutionStepVariable(Kratos.ACCELERATION)
+
+        number_of_nodes = 10
+        for id in range(1, number_of_nodes + 1):
+            node = cls.model_part.CreateNewNode(id, id, id+1, id+2)
+            node.SetValue(Kratos.PRESSURE, id+3)
+            node.SetValue(Kratos.VELOCITY, Kratos.Array3([id+3, id+4, id+5]))
+
+        number_of_conditions = 11
+        for id in range(1, number_of_conditions + 1):
+            properties = cls.model_part.CreateNewProperties(id)
+            properties.SetValue(Kratos.PRESSURE, id+400)
+            properties.SetValue(Kratos.VELOCITY, Kratos.Array3([id+500, id+600, id+700]))
+            condition = cls.model_part.CreateNewCondition("LineCondition2D2N", id + 1, [(id % number_of_nodes) + 1, ((id + 1) % number_of_nodes) + 1 ], properties)
+            condition.SetValue(Kratos.PRESSURE, id+4)
+            condition.SetValue(Kratos.VELOCITY, Kratos.Array3([id+5, id+6, id+7]))
+
+        number_of_elements = 12
+        for id in range(1, number_of_elements + 1):
+            properties = cls.model_part.CreateNewProperties(id + number_of_conditions)
+            properties.SetValue(Kratos.PRESSURE, id+500)
+            properties.SetValue(Kratos.VELOCITY, Kratos.Array3([id+600, id+700, id+800]))
+            element = cls.model_part.CreateNewElement("Element2D3N", id + 2, [(id % number_of_nodes) + 1, ((id + 1) % number_of_nodes) + 1, ((id + 2) % number_of_nodes) + 1 ], properties)
+            element.SetValue(Kratos.PRESSURE, id+5)
+            element.SetValue(Kratos.VELOCITY, Kratos.Array3([id+6, id+7, id+8]))
+
+        cls.utils = KratosOA.OptimizationUtils
+
+    def test_GetVariableDimension(self):
+        self.assertEqual(self.utils.GetVariableDimension(Kratos.PRESSURE, 1), 1)
+        self.assertEqual(self.utils.GetVariableDimension(Kratos.VELOCITY, 2), 2)
+
+    def test_IsVariableExistsInAllContainerProperties(self):
+        self.assertTrue(self.utils.IsVariableExistsInAllContainerProperties(self.model_part.Conditions, Kratos.PRESSURE, self.model_part.GetCommunicator().GetDataCommunicator()))
+        for condition in self.model_part.Conditions:
+            if condition.Id != 4:
+                condition.Properties[Kratos.DISTANCE] = 1.0
+        self.assertFalse(self.utils.IsVariableExistsInAllContainerProperties(self.model_part.Conditions, Kratos.DISTANCE, self.model_part.GetCommunicator().GetDataCommunicator()))
+
+    def test_IsVariableExistsInAtLeastOneContainerProperties(self):
```

Comment:
```suggestion
    def test_DoesVariableExistInAtLeastOneContainerProperty(self):
```
or "VariableExistInAtLeastOneContainerProperty(self)"
Same argument as before.

Reply:
same answer as before